debezium-embedded/src/test/java/io/debezium/embedded/EmbeddedEngineTest.java

```diff
@@ -149,6 +151,63 @@ public void shouldWorkToUseCustomChangeConsumer() throws Exception {
         stopConnector();
     }
 
+    @Test
+    public void shouldRunDebeziumEngine() throws Exception {
+        // Add initial content to the file ...
+        appendLinesToSource(NUMBER_OF_LINES);
+
+        final Properties props = new Properties();
+        props.setProperty("name", "debezium-engine");
+        props.setProperty("connector.class", "org.apache.kafka.connect.file.FileStreamSourceConnector");
+        props.setProperty(StandaloneConfig.OFFSET_STORAGE_FILE_FILENAME_CONFIG, OFFSET_STORE_PATH.toAbsolutePath().toString());
+        props.setProperty("offset.flush.interval.ms", "0");
+        props.setProperty("file", TEST_FILE_PATH.toAbsolutePath().toString());
+        props.setProperty("topic", "topicX");
+
+        CountDownLatch firstLatch = new CountDownLatch(1);
+        CountDownLatch allLatch = new CountDownLatch(6);
+
+        // create an engine with our custom class
+        final DebeziumEngine<SourceRecord> engine = DebeziumEngine.create(SourceRecord.class)
```

Comment:
So here's the one tiny concern I have: what will happen if we want to add support for other formats down the road (essentially by invoking converters under the hood)? E.g. what should be passed for obtaining JSON Strings? Back in the Reactive PoC we had an indirection for that purpose, something like:

```
interface SerializationFormat<T> {}

interface Json extends SerializationFormat<String> {}

interface Connect extends SerializationFormat<SourceRecord> {}
```

Then the usage could be like so:

```
// <T extends SerializationFormat<V>, V> DebeziumEngine<V> create(T format)
DebeziumEngine<SourceRecord> engine = DebeziumEngine.create(Connect.class)...
```

Might wanna do this now to be prepared for the evolution down the road.

Reply:
@gunnarmorling Indirect format definition added

Reply:
Ok, thx. LGTM.